pkg/virt-operator/resource/apply/apps.go

```diff
@@ -140,6 +145,20 @@ func (r *Reconciler) syncDaemonSet(daemonSet *appsv1.DaemonSet) error {
 	return nil
 }
 
+func setMaxDevices(kv *v1.KubeVirt, vh *appsv1.DaemonSet) {
+	maxDevices := getMaxDevices(kv)
+	if maxDevices == nil {
+		return
+	}
+
+	vh.Spec.Template.Spec.Containers[0].Command = append(vh.Spec.Template.Spec.Containers[0].Command, "--maxDevices")
+	vh.Spec.Template.Spec.Containers[0].Command = append(vh.Spec.Template.Spec.Containers[0].Command, fmt.Sprintf("%d", *maxDevices))
+}
+
+func getMaxDevices(kv *v1.KubeVirt) *int {
+	return kv.Spec.Configuration.MaxDevices
+}
```

Comment:
I'm normally in favor of abstraction and extracting logic to separate methods, but I feel this particular case is somewhat redundant.
I think the following is perfectly fine:
```go
if kv.Spec.Configuration.MaxDevices == nil {
	return
}
command = append(command *kv.Spec.Configuration.MaxDevices)
```